### PR TITLE
[CBRD-24111] Problem with view merging when there is a variable definition ( ':=' ) in the where clause.

### PR DESCRIPTION
**Pull request by shparkcubrid**
http://jira.cubrid.org/browse/CBRD-24111

if the variable definition(:=) is in the WHERE of the main query, I fix so that view merge is not performed.

### CODE REVIEW

File: src/parser/view_transform.c [hunk 1]
```diff
@@ -1735,7 +1736,7 @@ mq_substitute_subquery_in_statement (PARSER_CONTEXT * parser, PT_NODE * statemen
 	  rewrite_as_derived = true;
 	}
       /* check for CONNECT BY */
-      else if (tmp_result->info.query.q.select.connect_by)
+      else if (PT_IS_SELECT (tmp_result) && tmp_result->info.query.q.select.connect_by)
```
Comment (beyondykk9) [inline src/parser/view_transform.c:1739]: it looks not related to this issue.

Comment (shparkcubrid) [reply src/parser/view_transform.c:1739]: There was a problem with node type and I fixed it together. :)

File: src/parser/parser_support.c [hunk 2]
```diff
@@ -2946,6 +2946,54 @@ pt_has_aggregate (PARSER_CONTEXT * parser, PT_NODE * node)
...
+ * arg (in)    :
+ * continue_walk (in) :
+ */
+PT_NODE *
+pt_is_define_vars (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk)
+{
+  bool *is_define_vars = (bool *) arg;
+  *continue_walk = PT_CONTINUE_WALK;
+
+  if (*is_define_vars)
```
Comment (hgryoo) [inline src/parser/parser_support.c:2980]:
```suggestion
  if (is_define_vars && *is_define_vars)
```

**Review by hgryoo — APPROVED** [review]

File: src/parser/parser_support.c (hunk 2 above)
Comment (ctshim) [reply src/parser/parser_support.c:2980]: In addition to @hgryoo's comment, how about modifying it as follows?
```
 if( is_define_vars == NULL || *is_define_vars)
 {
      /* stop checking, there already is a parameter in the statement */
      *continue_walk = PT_STOP_WALK;
      return node;
    }
```

Comment (shparkcubrid) [reply src/parser/parser_support.c:2980]: I'll put an assert.
```
assert (has_define_vars != NULL)
```

**Review by ctshim — APPROVED** [review]

**Review by beyondykk9 — APPROVED** [review]

File: src/parser/parser_support.c [hunk 3]
```diff
@@ -2946,6 +2946,55 @@ pt_has_aggregate (PARSER_CONTEXT * parser, PT_NODE * node)
...
+ * node (in)   : node
+ * arg (in)    :
+ * continue_walk (in) :
+ */
+PT_NODE *
+pt_is_define_vars (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue_walk)
+{
+  bool *is_define_vars = (bool *) arg;
+  *continue_walk = PT_CONTINUE_WALK;
+  assert (has_define_vars != NULL);
```
Comment (youngjinj) [inline src/parser/parser_support.c]: **has_define_vars** is a typo. should be changed to **is_define_vars**.

File: src/parser/parser_support.c (hunk 2 above)
Comment (youngjinj) [reply src/parser/parser_support.c:2980]: **has_define_vars** is a typo. should be changed to **is_define_vars**.

Comment (youngjinj) [reply src/parser/parser_support.c:2980]: For pointer variables, I think it is safe to always check the address first and then check the value.
Because assert works only in debug mode, it is difficult to check in release mode.
For this reason, I agree with hgryoo's opinion.

Please consider it as one opinion.

**Review by youngjinj — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

File: src/parser/parser_support.c (hunk 2 above)
Comment (shparkcubrid) [reply src/parser/parser_support.c:2980]: I fixed it. thanks.

File: src/parser/parser_support.c (hunk 3 above)
Comment (shparkcubrid) [reply src/parser/parser_support.c]: I fixed it.
